src/Security/MemberAuthenticator/MemberAuthenticator.php

```diff
@@ -53,68 +57,108 @@ public function authenticate($data, &$result = null)
+            $member->validateCanLogin($result);
+            if ($result->isValid()) {
+                // Check if default admin credentials are correct
+                if (DefaultAdminService::isDefaultAdminCredentials($email, $data['Password'])) {
+                    return $member;
+                } else {
+                    $result->addError(_t(
+                        'SilverStripe\\Security\\Member.ERRORWRONGCRED',
+                        "The provided details don't seem to be correct. Please try again."
+                    ));
+                }
             }
         }
 
         // Attempt to identify user by email
         if (!$member && $email) {
             // Find user by email
+            $identifierField = Member::config()->get('unique_identifier_field');
             /** @var Member $member */
             $member = Member::get()
-                ->filter([Member::config()->get('unique_identifier_field') => $email])
+                ->filter([$identifierField => $email])
                 ->first();
         }
 
         // Validate against member if possible
         if ($member && !$asDefaultAdmin) {
-            $result = $member->checkPassword($data['Password']);
+            $this->checkPassword($member, $data['Password'], $result);
         }
 
         // Emit failure to member and form (if available)
         if (!$result->isValid()) {
             if ($member) {
                 $member->registerFailedLogin();
             }
+        } elseif ($member) {
+            $member->registerSuccessfulLogin();
         } else {
-            if ($member) {
-                $member->registerSuccessfulLogin();
-            } else {
-                // A non-existing member occurred. This will make the result "valid" so let's invalidate
-                $result->addError(_t(
-                    'SilverStripe\\Security\\Member.ERRORWRONGCRED',
-                    "The provided details don't seem to be correct. Please try again."
-                ));
-                $member = null;
-            }
+            // A non-existing member occurred. This will make the result "valid" so let's invalidate
+            $result->addError(_t(
+                'SilverStripe\\Security\\Member.ERRORWRONGCRED',
+                "The provided details don't seem to be correct. Please try again."
+            ));
+            return null;
         }
 
         return $member;
     }
 
     /**
+     * Check if the passed password matches the stored one (if the member is not locked out).
+     *
+     * Note, we don't return early, to prevent differences in timings to give away if a member
+     * password is invalid.
+     *
+     * @param Member $member
+     * @param string $password
+     * @param ValidationResult $result
+     * @return ValidationResult
+     */
+    public function checkPassword(Member $member, $password, ValidationResult $result = null)
```

Comment:
It's also passed into validateCanLogin() before being overwritten, so it's safe.